packages/@lwc/ssr-compiler/src/compile-js/index.ts

```diff
@@ -81,6 +89,12 @@ const visitors: Visitors = {
             return;
         }
 
+        // If we mutate any class-methods that are piped through this compiler, then we'll be
+        // inadvertently mutating things like Wire adapters.
+        if (!state.isLWC) {
+            return;
+        }
```

Comment:
I lost a few minutes trying to figure out what was going wrong here. Pretty sure this is the right thing to do in this case, and you might consider doing similar checks in other places where the AST is being mutated.

Reply:
Just noticed this doesn't work for superclasses. I'll open an issue for this.

Reply:
Yeah the logic for determining `isLWC` definitely needs some work.

Reply:
- https://github.com/salesforce/lwc/issues/4818